tests/e2e/scripts/run_tests.sh

```diff
@@ -1,62 +1,60 @@
 #!/bin/bash
-nodeOS=${1:-"generic/ubuntu2004"}
-servercount=${2:-3}
-agentcount=${3:-1}
-db=${4:-"etcd"}
-hardened=${5:-""}
-rke2_version=${rke2_version}
-rke2_channel=${rke2_channel:-"commit"}
-
-E2E_EXTERNAL_DB=$db && export E2E_EXTERNAL_DB
+# Usage: ./run_tests.sh
+# This script runs all the rke2 e2e tests and generates a report with the log
+# The generated log is placed in createreport/rke2_${date}.log
+#
+# This script must be run inside the rke2 directory where the tests exist
+#
+# Example:
+#   To run the script with default settings:
+#     ./run_tests.sh
+#
+set -x
+
+# tests to run
+tests=("ciliumnokp" "dnscache" "dualstack" "mixedos" "mixedosbgp" "multus" "secretsencryption" "splitserver" "upgradecluster" "validatecluster")
+date=$(date +%Y%m%d)
+
 E2E_REGISTRY=true && export E2E_REGISTRY
 
-cd
-cd rke2 && git pull --rebase origin master
+git pull --rebase origin master
 /usr/local/go/bin/go mod tidy
-
 cd tests/e2e
-OS=$(echo "$nodeOS"|cut -d'/' -f2)
-echo "$OS"
 
 # create directory to store reports if it does not exists
 if [ ! -d createreport ]
 then
 	mkdir createreport
 fi
 
+# Remove VMs which are in invalid state
+vagrant global-status --prune
+
 count=0
 run_tests(){
-	count=$(( count + 1 ))
-	vagrant global-status | awk '/running/'|cut -c1-7| xargs -r -d '\n' -n 1 -- vagrant destroy -f
-
-	E2E_RELEASE_VERSION=$rke2_version && export E2E_RELEASE_VERSION
-	E2E_RELEASE_CHANNEL=$rke2_channel && export E2E_RELEASE_CHANNEL
-
-	echo 'RUNNING CLUSTER UPGRADE TEST'
-	E2E_REGISTRY=true /usr/local/go/bin/go test -v ./upgradecluster/upgradecluster_test.go -nodeOS="$nodeOS" -serverCount=$((servercount)) -agentCount=$((agentcount)) -timeout=1h -json -ci |tee createreport/rke2_"$OS".log
-
-	echo 'RUNNING DUALSTACK VALIDATION TEST'
-	E2E_HARDENED="$hardened" /usr/local/go/bin/go test -v dualstack/dualstack_test.go -nodeOS="$nodeOS" -serverCount=1 -agentCount=1  -timeout=30m -json -ci |tee -a createreport/rke2_"$OS".log
 
-	echo 'RUNNING CLUSTER VALIDATION TEST'
-	E2E_REGISTRY=true E2E_HARDENED="$hardened" /usr/local/go/bin/go test -v validatecluster/validatecluster_test.go -nodeOS="$nodeOS" -serverCount=$((servercount)) -agentCount=$((agentcount))  -timeout=30m -json -ci |tee -a createreport/rke2_"$OS".log
+    count=$(( count + 1 ))
 
-	echo 'RUNNING MIXEDOS TEST'
-	/usr/local/go/bin/go test -v mixedos/mixedos_test.go -nodeOS="$nodeOS" -serverCount=$((servercount)) -timeout=1h -json -ci |tee -a  createreport/rke2_"$OS".log
+    for i in ${!tests[@]}; do
```

Comment:
Some tests have just server nodes and no agents. Some have HA and some single server single agent. Can we default then in Vagrantfile?

Reply:
Yes, vagrantfile defaults the number of servers and agents per test. Same for the OS